pkg/controller/clusterdeployment/clusterdeployment_controller.go

```
@@ -1309,3 +1340,111 @@ func initializeAnnotations(cd *hivev1.ClusterDeployment) {
 		cd.Annotations = map[string]string{}
 	}
 }
+
+// mergePullSecrets takes pull secretes in to account and returns the merged pull request
+func (r *ReconcileClusterDeployment) mergePullSecrets(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (string, error) {
+
+	// For code readability let's call the pull secret in cluster deployment config as local pull secret
+	localPullSecret, _ := controllerutils.LoadSecretData(r.Client, cd.Spec.PullSecret.Name, cd.Namespace, corev1.DockerConfigJsonKey)
```

Do not ignore the error return.